packages/astro/src/core/build/generate.ts

```diff
@@ -631,6 +629,19 @@ async function createBuildManifest(
 			domainLookupTable: {},
 		};
 	}
+
+	if (shouldTrackCspHashes(settings.config)) {
+		const algorithm = getAlgorithm(settings.config);
+		const clientScriptHashes = await trackScriptHashes(internals, settings, algorithm);
+		const clientStyleHashes = await trackStyleHashes(internals, settings, algorithm);
+
+		csp = {
+			clientStyleHashes,
+			clientScriptHashes,
+			shouldInjectCspMetaTags: shouldTrackCspHashes(settings.config),
```

Comment:
I don't think it makes sense to have this flag as part of the SSRManifestCSP, since this manifest is only created if `shouldTrackCspHashes(settings.config)`

Reply:
We must have this flag, because in SSR we don't have a configuration anymore, so we must rely on the manifest to hold this information.